### PR TITLE
Port Scaler to TypeScript

### PR DESCRIPTION
**Pull request by evancharlton**
Bring the world of typing to Scaler. This introduces some TODOs because
of some types which need to live in yet-to-be-ported components (eg,
DataProvider). These will be cleaned up when that migration happens.

### CODE REVIEW

**Comment by cognite-cicd** [comment]
[pr-server]
The storybook for this PR is hosted on https://griff-react-pr-285.surge.sh

File: src/components/Scaler/index.tsx [hunk 1]
```diff
@@ -43,13 +84,23 @@ const findItemsWithChangedDomains = (previousItems, currentItems) => {
...
 
-export const stripPlaceholderDomain = domain => {
+export const stripPlaceholderDomain = (domain: Domain): Domain | undefined => {
   if (isEqual(PLACEHOLDER_DOMAIN, domain)) {
     return undefined;
   }
   return domain;
 };
 
+const isEqual = (a: Domain, b: Domain): boolean => {
```
Comment (valerii-cognite) [inline src/components/Scaler/index.tsx]: What's up with `lodash.isequal`?

Comment (evancharlton) [reply src/components/Scaler/index.tsx]: Dunno, TypeScript was unhappy with it and I realized we're only using it to compare two Domains -- we can reimplement that pretty easily. 🤷‍♂️

**Comment by cognite-cicd** [comment]
[pr-server]
The storybook for this PR is hosted on https://griff-react-pr-285.surge.sh

**Pull request merged**